### PR TITLE
Fix root device overlapping

### PR DESCRIPTION
**Pull request by SchSeba**

### CODE REVIEW

File: api/v1/helper.go [hunk 1]
```diff
@@ -547,6 +547,19 @@ func ParsePFName(name string) (ifName string, rngSt, rngEnd int, err error) {
 	return
 }
 
+// ParseRootDevice: partse the rootDevice with VF range
```
Comment (e0ne) [inline api/v1/helper.go:550]: nit: partse => parse

Comment (SchSeba) [reply api/v1/helper.go]: done thanks!

File: api/v1/helper.go [hunk 2]
```diff
@@ -481,7 +481,7 @@ func (p *SriovNetworkNodePolicy) generateVfGroup(iface *InterfaceExt) (*VfGroup,
 	var rngStart, rngEnd int
 	found := false
 	for _, selector := range p.Spec.NicSelector.PfNames {
```
Comment (zeeke) [inline api/v1/helper.go:483]: Ranges are still defined on `PfNames` field only. 
You can verify it by adding unit tests like  https://github.com/k8snetworkplumbingwg/sriov-network-operator/commit/a92b76433a5debe9749e82a66ebcb35536b2453e .

Comment (SchSeba) [reply api/v1/helper.go:483]: you are right!

I think supporting rage in the root device will be much harder than I was expecting so I will first fix the webhook issue and than we can take to the community to talk about the range support for rootDevices.

let me update the PR

**Review by zeeke — APPROVED** [review]
LGTM

**Comment by SchSeba** [comment]
Hi @e0ne @Eoghan1232 when you have time please take a look on this small PR

**Review by Eoghan1232 — APPROVED** [review]

**Comment by SchSeba** [comment]
Thanks for the review folks!

merging this one

**Pull request merged**